crypto/cmp/cmp_asn.c

```diff
@@ -332,6 +351,243 @@ int OSSL_CMP_ITAV_get0_rootCaKeyUpdate(const OSSL_CMP_ITAV *itav,
     return 1;
 }
 
+OSSL_CMP_ITAV
+*OSSL_CMP_ITAV_new0_crlStatusList(STACK_OF(OSSL_CMP_CRLSTATUS) *crlStatusList)
+{
+    OSSL_CMP_ITAV *itav;
+
+    if ((itav = OSSL_CMP_ITAV_new()) == NULL)
+        return NULL;
+    itav->infoType = OBJ_nid2obj(NID_id_it_crlStatusList);
+    itav->infoValue.crlStatusList = crlStatusList;
+    return itav;
+}
+
+int OSSL_CMP_ITAV_get0_crlStatusList(const OSSL_CMP_ITAV *itav,
+                                     STACK_OF(OSSL_CMP_CRLSTATUS) **out)
+{
+    if (itav == NULL || out == NULL) {
+        ERR_raise(ERR_LIB_CMP, ERR_R_PASSED_NULL_PARAMETER);
+        return 0;
+    }
+    if (OBJ_obj2nid(itav->infoType) != NID_id_it_crlStatusList) {
+        ERR_raise(ERR_LIB_CMP, ERR_R_PASSED_INVALID_ARGUMENT);
+        return 0;
+    }
+    *out = itav->infoValue.crlStatusList;
+    return 1;
+}
+
+OSSL_CMP_CRLSTATUS *OSSL_CMP_CRLSTATUS_new1(const DIST_POINT_NAME *dpn,
+                                            const GENERAL_NAMES *issuer,
+                                            const ASN1_TIME *thisUpdate)
+{
+    OSSL_CMP_CRLSOURCE *crlsource;
```

Comment:
```suggestion
    OSSL_CMP_CRLSOURCE *crlsource = NULL;
```

Reply:
Unnecessary as these are assigned a few lines below.

Reply:
@t8m - I understand that these variables are assigned values a few lines below. However, I'm uncertain why we're resistant to adhering to best practices in our code base. Don't you agree that initializing variables is a fundamental aspect of writing clean and maintainable code?

By following this best practice, we can significantly reduce the likelihood of bugs and enhance code clarity. Explicitly initializing variables communicates our intentions clearly and improves readability for anyone who reviews our code, including ourselves in the future.

Moreover, consistency in our coding standards establishes a benchmark for future generations of developers to follow. It sets a standard of excellence that fosters better collaboration, code quality, and maintainability.

Would you be open to discussing this further and considering the benefits of adopting best practices in our code base?

Reply:
I disagree that initializing variables that are being assigned values immediately after is a good practice and improves the readability and maintainability of the code. It adds just an unnecessary visual cruft and with some static analyzers it even gives warnings that the value is unconditionally overwritten.

However regardless of whether I agreed with you or not, such code style changes should be first formally agreed and approved by OTC, before they are required by code reviewers. So it is not matter of a personal preference for a particular code style but consistent requirements applied throughout the code base.